app/controllers/course/forum/topics_controller.rb

```diff
@@ -9,8 +9,10 @@ class Course::Forum::TopicsController < Course::Forum::ComponentController
   before_action :add_topic_breadcrumb
 
   def show
-    @posts = @topic.posts
+    @posts = @topic.posts.order(created_at: :asc).includes(:creator).to_a
     @topic.views.create(user: current_user)
+    @post = @topic.acting_as.posts.build(title: t('course.discussion.posts.reply_title',
```

Comment:
I think no special reason, using `@topic.posts` will also work. I'm using this just for making sure it is built from the discussion_topic. And this may have less potential problem.